mreg_cli/api/models.py

```diff
@@ -256,6 +256,42 @@ def valid_numeric_ttl(self, ttl: int) -> int:
         return ttl
 
 
+class WithName(APIMixin[Any]):
```

Comment:
Does it make any realistic difference to use BaseModel here instead?

Reply:
Nope, I can change it.

Reply:
Nevermind, it does matter: 

```
Base classes of Atom are mutually incompatible
  Base class "APIMixin[Atom]" is incompatible with type "APIMixin[BaseModel]"
```

I don't love using `Any` as the generic type, and it is an indication that there is a deeper architectural issue. However, `WithName.get_by_name` gives us the correct return type because of its `Self` annotation, even when using `Any` as the generic type.

Reply:
I find "deeper architectural issue" to be a funny way of saying "python, are you okay"? :)

`Any` works. There is only so much and so deep meta programming we want to do. This isn't something python does well to begin with and doing with the expectation of semi-strong typing doesn't increase the odds of it going well.

Reply:
I suggest we use this as a stepping stone into a larger refactoring down the line. The type safety of `WithName` doesn't matter much as long as it's used alongside a `BaseModel` type and a `APIMixin` with a concrete type argument (which we always do).

Reply:
Sounds good. Although, er, spanner, works, etc... In some cases, the `name` field is of the type HostT (Cname and Host), in other cases, it's a str. Funnily enough (no, not really), in BacnetID, the field is `hostname`. :)

Reply:
> In some cases, the name field is of the type HostT (Cname and Host), in other cases, it's a str. 

Yeah, but we actually don't directly access the attribute; the methods are classmethods that expect a string argument. It is the job of the caller to pass in a string type, which Pylance will catch if done incorrectly.

> Funnily enough (no, not really), in BacnetID, the field is hostname. :)

THIS is a problem though

Reply:
> > Funnily enough (no, not really), in BacnetID, the field is hostname. :)
>
> THIS is a problem though

Do we want to alias hostname in BacnetID? And will that work?